src/Graphics/ColorBufferReader.cpp

```diff
@@ -37,15 +38,15 @@ namespace graphics {
 	}
 
 	const u8* ColorBufferReader::_convertIntegerTextureBuffer(const u8* _gpuData, u32 _width, u32 _height,
-		u32 _heightOffset, u32 _stride)
+		u32 _heightOffset, u32 _stride, u32 _colorsPerPixel)
 	{
-		const u32 colorsPerPixel = 4;
-		const u32 widthBytes = _width * colorsPerPixel;
-		const u32 strideBytes = _stride * colorsPerPixel;
+		const u32 widthBytes = _width * _colorsPerPixel;
+		const u32 strideBytes = _stride * _colorsPerPixel;
 
 		u8* pixelDataAlloc = m_pixelData.data();
-		for (u32 lnIndex = 0; lnIndex < _height; ++lnIndex) {
-			memcpy(pixelDataAlloc + lnIndex*widthBytes, _gpuData + ((lnIndex+_heightOffset)*strideBytes), widthBytes);
+
+		for (u32 index = 0; index < _height && (index * widthBytes + widthBytes < m_pixelData.size()); ++index) {
```

Comment:
Same as above: reduce _height to avoid out of bounds writes.

Reply:
If it matters still,I went through a fresh file before on mupen Android and still encountered the issue with the entire process being without any cheats. I thought that I mentioned this already to @fzurita though? It was a native save too,no save states loaded in that lengthy test. I went out of my own way to replay a cleanly made save up to that point just to prove its a normal issue of the emulation or whatever part is actually triggering it.

Reply:
I think you did mention it, but I forgot.

Reply:
>> I went through a fresh file before on mupen Android and still encountered the issue with the entire process being without any cheats

So, it is not caused by cheats. Bad news. @fzurita @loganmc10  could you check, is this bug reproduced with other graphics plugins? I mean not crash in that point, but bad set of tile parameters, which leads to crash in load tlut: tile.tmem > 256 and tile.lrs == 255. It can be checked with debugger or with logging.
If it is reproduced, it is core issue. Otherwise, the plugin somehow breaks things.

Reply:
I have not experienced any crashing with Glide64mk2, but I have not checked that we are getting bad tile parameters. I'll try to check that when I get a chance, I have been busy with other things last couple of weeks.

Reply:
So just to confirm a few things, @gonetz's fix does prevent the crash. I tested Glide64mk2 and it doesn't crash. I can test Glide64mk2 for the values:

Can someone point out to me where in https://github.com/mupen64plus/mupen64plus-video-glide64mk2/ I should be looking for these values? I'm not familiar with the source code at all

Reply:
I think I found it here:
https://github.com/mupen64plus/mupen64plus-video-glide64mk2/blob/master/src/Glide64/rdp.cpp#L1693-L1696

I added:
```
if (start != 0 && count != 16)
      printf("start: %u count: %u\n", start, count);
```
Right where that crash happens it prints out a few:
```
start: 64 count: 256
```
It seems to be the only place it happens. So yes, it does seem that the same issue happens in Glide64, just no crash.

Reply:
@loganmc10 Yes, start: 64 count: 256 is violation of "if palette index != 0 palette size is 16" invariant.  Thanks! So, this is most likely a core issue. May be playing with counter factor can help. IIRC, Banjo-Tooie is very sensitive to counter factor and timing issues.

Reply:
Ok, it seems like we have established that bad tile parameters are a real possibility in mupen64plus and probably project64 as well. Could a custom microcode cause this issue?

Reply:
No, gDPSetTile and gDPSetTileSize are RDP commands. I did not check that gDPSetTileSize is actually missing, May be it presents, but sets wrong data. In both cases it is an error in display list, emulator's fail. Btw, have you checked that save in LLE mode? I can't check it now as I don't have Linux box on hands.  And the crash does not happen on windows for me.